core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1247,7 +1247,8 @@ class ReplicaManager(val config: KafkaConfig,
         val usableBytes = adjustForLargeFileSystems(fileStore.getUsableSpace)
         logsByDir.get(absolutePath) match {
           case Some(logs) =>
-            val topicInfos = logs.groupBy(_.topicPartition.topic).map{case (topic, logs) =>
+            val topicInfos = logs.filter(_.topicPartition.topic.nonEmpty)
```

Comment:
I don't think the bug is due to `UnifiedLog.topicPartition.topic` being empty but rather because we don't filter for logs which pertain to the topics in the argument to the function.

Consider the case where the `logs` refers to logs for `["topicA-0", "topicA-1", "topicB-0", "topicB-1"]` and `partitions` is `["topicA-0", "topicA-1"]`

Reply:
agree to @gaurav-narula 

Maybe we can just add a filter to remove the topics having "zero" partitions.

Reply:
@chia7712 , @gaurav-narula , Thanks for your comments, I changed the code